diff --git a/dist/server/image-optimizer.js b/dist/server/image-optimizer.js
index 7c558bc8bb3fff0d9d77c8d8ced6d510b3fe01f1..cf7783952869174fc564b591153a76fcd87ee628 100644
--- a/dist/server/image-optimizer.js
+++ b/dist/server/image-optimizer.js
@@ -1017,7 +1017,8 @@ async function fetchInternalImage(href, _req, _res, handleRequest) {
         const mocked = (0, _mockrequest.createRequestResponseMocks)({
             url: href,
             method,
-            socket: _req.socket
+            socket: _req.socket,
+            headers: _req.headers
         });
         await handleRequest(mocked.req, mocked.res, _url.default.parse(href, true));
         await mocked.res.hasStreamed;
